sklearn/utils/estimator_checks.py

```diff
@@ -398,10 +399,11 @@ def check_dtype_object(name, Estimator):
 
 
 @ignore_warnings
-def check_dict(name, Estimator):
+def check_dict_unchanged(name, Estimator):
+    print(name)
     rnd = np.random.RandomState(0)
-    if name in ['SpectralCoclustering']:
-        return 0
+    if name in ['SpectralCoclustering', 'NMF', 'ProjectedGradientNMF']:
```

Comment:
How do they change the dict? Please leave them in so we can see the error and fix them.